File: Src/Icm.Core/IO/DateStringWriter.vb
Imports System.IO

Namespace Icm.IO

    Public Class DateStringWriter
        Inherits StringWriter

        Public Sub New()
            MyBase.New(CultureInfo.InvariantCulture)
        End Sub

        Public Overloads Overrides Sub WriteLine(ByVal value As String)
            MyBase.WriteLine("{0:dd/MM/yyyy HH:mm:ss} {1}", Now, value)
        End Sub

    End Class

End Namespace

File: Src/Icm.Core/IO/DateStringWriter.cs
using System;
using System.Globalization;
using System.IO;

namespace Icm.IO
{

	public class DateStringWriter : StringWriter
	{

		public DateStringWriter() : base(CultureInfo.InvariantCulture)
		{
		}

		public override void WriteLine(string value)
		{
			base.WriteLine("{0:dd/MM/yyyy HH:mm:ss} {1}", DateTime.Now, value);
		}

	}

}